static/js/containers/BookDetailPage.js

```diff
@@ -0,0 +1,80 @@
+// @flow
+import React from "react"
+import R from "ramda"
+import { connect } from "react-redux"
+import {
+  connectRequest,
+  querySelectors,
+} from "redux-query"
+import { Link } from "react-router-dom"
+
+import { bookRequest } from "../lib/api"
+import { updateBook } from "../actions"
+
+class BookDetailPage extends React.Component<*, void> {
+  state = {
+    titleEdit: ""
+  }
+
+  onTitleEditChange = (e) => {
+    this.setState({
+      titleEdit: e.target.value,
+    });
+  }
+
+  onSubmit = () => {
+    const { titleEdit } = this.state
+    if (titleEdit === "") {
+      return
+    }
+    this.props.updateBook(
+      this.props.book.id,
+      {title: this.state.titleEdit}
+    )
+  }
+
+  render() {
+    const { isLoading, book } = this.props
+
+    return (
+      <div>
+        <Link to={"/library"}>back</Link><br />
+        {isLoading && <div><em>Loading...</em></div>}
+        {book && (
+          <div>
+            <div>Author: {book.author.name}</div>
+            <div>Title: {book.title}</div>
+            <div>
+              New title: <input type="text" value={this.state.titleEdit} onChange={this.onTitleEditChange} />
+            </div>
+            <div><button onClick={this.onSubmit}>Submit</button></div>
+          </div>
+        )}
+      </div>
+    )
+  }
+}
+
+const mapStateToProps = (state, ownProps) => {
+  const bookQuery = bookRequest(ownProps.match.params.bookId)
+  return {
+    isLoading: querySelectors.isPending(state.queries, bookQuery),
+    book:      state.entities.book,
+    query:     bookQuery
+  }
+}
+
+const mapPropsToConfigs = (props) => props.query
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    updateBook: (bookId, payload) => {
+      dispatch(updateBook(bookId, payload))
+    },
+  }
+}
+
+export default R.compose(
+  connect(mapStateToProps, mapDispatchToProps),
+  connectRequest(mapPropsToConfigs)
```

Comment:
[`connectRequest`](https://github.com/amplitude/redux-query#connectrequest) is basically the way you express that a component needs some data when it mounts. By default there will _not_ be a request made if some data has already been loaded for the endpoint, but if it should be freshly loaded each time the component mounts, you can add the `force` flag to the query config object ([demonstrated here](https://amplitude.github.io/redux-query/#/mounting) in one of their examples)